docs/csharp/csharp-6.md

```diff
@@ -1,17 +1,570 @@
+
+[!code-csharp[ExceptionFilterLogging](../../samples/snippets/csharp/new-in-6/ExceptionFilterHelpers.cs#ExceptionFilterLogging)]
+
+Whenever you want to log an exception, you can add a catch clause, and
+use this method as the exception filter:
+
+[!code-csharp[LogException](../../samples/snippets/csharp/new-in-6/program.cs#LogException)]
+
+The exceptions are never caught, because the `LogException` method always
+returns `false`. That always false exception filte means that you can
+place this logging handler before any other exception handlers:
+
+[!code-csharp[LogExceptionRecovery](../../samples/snippets/csharp/new-in-6/program.cs#LogExceptionRecovery)]
+
+The above example highlights a very important facet of exception filters.
+The exception filters enable scenarios where a more general exception
+catch clause may appear before a more specific one. It's also possible
+to have the same exception type appear in multiple catch clauses:
+
+[!code-csharp[HandleNotChanged](../../samples/snippets/csharp/new-in-6/NetworkClient.cs#HandleNotChanged)]
+
+Another recommended pattern helps prevent catch clauses from processing
+exceptions when a debugger is attached. This technique enables you to
+run an application with the debugger, and stop execution when an exception
+is thrown.
+
+In the code, add an exception filter so that any recovery code executes
+only when a debugger is not attached:
+
+[!code-csharp[LogExceptionDebugger](../../samples/snippets/csharp/new-in-6/program.cs#LogExceptionDebugger)]
+
+After adding this in code, you set your debugger to break on all unhandled
+exceptions. Run the program under the debugger, and the debugger breaks
+whenever `PerformFailingOperation()` throws a `RecoverableException`.
+The debugger breaks your program, because the catch clause won't be executed
+due to the false-returning exception filter.
+
+## `nameof` Expressions
+
+The `nameof` expression evaluates to the name of a symbol. It's a great
+way to get tools working with you whenever you need the name of a variable,
+a property, or a member field.
+
+One of the most common uses for `nameof` is to provide the name of a symbol
+that caused an exception:
+
+[!code-csharp[nameof](../../samples/snippets/csharp/new-in-6/NewCode.cs#UsingStaticString)]
+
+Another use is with XAML based applications that implement the `INotifyPropertyChanged`
+interface:
+
+[!code-csharp[nameofNotify](../../samples/snippets/csharp/new-in-6/viewmodel.cs#nameofNotify)]
+
+The advantage of using the `nameof` operator over a constant string is
+that tools can understand the symbol. If you use refactoring tools to
+rename the symbol, it will rename it in the `nameof` expression. Constant
+strings don't have that advantage. Try it yourself in your favorite editor:
+Rename a variable, and any nameof expressions will update as well.
+
+The `nameof` expression produces the unqualified name of its argument
+(`LastName` in the examples above) even if you use the fully qualified
```

Comment:
examples above -> previous example (because the first example used nameof(lastName) and not LastName